lib/pbench/server/api/resources/quisby.py

```diff
@@ -0,0 +1,104 @@
+from http import HTTPStatus
+from urllib.request import Request
+
+from flask import current_app, jsonify
+from flask.wrappers import Response
+from pquisby.lib.post_processing import BenchmarkName, InputType, QuisbyProcessing
+
+from pbench.server import OperationCode, PbenchServerConfig
+from pbench.server.api.resources import (
+    APIAbort,
+    ApiAuthorizationType,
+    ApiBase,
+    ApiContext,
+    APIInternalError,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+    Parameter,
+    ParamType,
+    Schema,
+)
+from pbench.server.cache_manager import (
+    CacheManager,
+    TarballNotFound,
+    TarballUnpackError,
+)
+from pbench.server.database import Dataset
+
+
+class QuisbyData(ApiBase):
+    """
+    API class to retrieve Quisby data for a dataset
+    """
+
+    def __init__(self, config: PbenchServerConfig):
+        super().__init__(
+            config,
+            ApiSchema(
+                ApiMethod.GET,
+                OperationCode.READ,
+                uri_schema=Schema(
+                    Parameter("dataset", ParamType.DATASET, required=True),
+                ),
+                authorization=ApiAuthorizationType.DATASET,
+            ),
+        )
+
+    def _get(
+        self, params: ApiParams, request: Request, context: ApiContext
+    ) -> Response:
+        """
+        This function returns the Quisby data for the requested dataset.
+
+        Args:
+            params: includes the uri parameters, which provide the dataset.
+            request: Original incoming Request object
+            context: API context dictionary
+
+        Raises:
+            APIAbort, reporting "NOT_FOUND" and "INTERNAL_SERVER_ERROR"
+
+        GET /api/v1/quisby/{dataset}
+        """
+
+        dataset = params.uri["dataset"]
+
+        cache_m = CacheManager(self.config, current_app.logger)
+        try:
+            tarball = cache_m.find_dataset(dataset.resource_id)
+        except TarballNotFound as e:
+            raise APIAbort(HTTPStatus.NOT_FOUND, str(e))
+
+        name = Dataset.stem(tarball.tarball_path)
+        try:
+            file = tarball.extract(tarball.tarball_path, f"{name}/result.csv")
+        except TarballUnpackError as e:
+            raise APIInternalError(str(e)) from e
+
+        metadata = self._get_dataset_metadata(
+            dataset, ["dataset.metalog.pbench.script"]
+        )
+        benchmark = metadata["dataset.metalog.pbench.script"]
+        if benchmark == "uperf":
+            benchmark_type = BenchmarkName.UPERF
+        elif benchmark == "fio":
+            benchmark_type = BenchmarkName.FIO
```

Comment:
Are you actually supporting "fio" at this point? I thought the description said "only uperf"?

Also, for general information, we can identify a "fio" `pbench-user-benchmark` run by looking at `dataset.metalog.iterations/1-default.user_script`, which is the user command (after the `--`) separated into tokens by `,`. That is, something like `"user_script": "fio, --directory=/fiotest, --name, fio_test_file, --direct=1, --rw=randread, --bs=16k, --size=100M, --numjobs=8, --time_based, --runtime=5s, --group_reporting, --norandommap"`. So `v.split(",")[0]` should be "fio" (or "uperf" or whatever).

A few complications we'll need to work through (I'm definitely *not* suggesting tackling it in this PR):

1. Technically the `1-default` is a default iteration name, but with `--iterations` they can call it whatever they want; so we'd probably need to get `dataset.metalog` and look for keys with the "iterations/<name>" pattern.
2. Actually, the first complication may simplify this one: much of the metadata processing code assumes that metadata keys are lowercase alphanumeric, and checks: obviously the `1-default` violates that. The datasets list filter and metadata processing has special code to allow access to the weirder Pbench Agent metadata.log keys (many are much weirder than this), but we need to be careful. On the other hand, `dataset.metalog` will be fine and maybe we won't even need to worry about it.